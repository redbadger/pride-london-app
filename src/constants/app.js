// @flow
export const appNameIos = "pride-in-london";
export const appIdIos = "1250496471";
