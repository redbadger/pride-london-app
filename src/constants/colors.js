export const bgColor = "rgb(255, 255, 255)";
export const cardBgColor = "rgb(255, 255, 255)";
export const imageBgColor = "#C1C1C1";
export const headerBgColor = "#A5A5A5";

export const textColor = "#272727";

export const dialogBackdropColor = "rgba(0, 0, 0, 0.7)";
export const dialogBgColor = "#FFFFFF";
export const dialogApplyButtonBgColor = "rgb(51, 51, 51)";
export const dialogApplyButtonTextColor = "#FFFFFF";
export const dialogHeaderDividerColor = "#F2F2F2";

// FILTER HEADER

export const filterBgColor = "rgb(45, 47, 127)";
export const interestButtonBgColor = "rgb(32, 33, 90)";
export const interestButtonTextColor = "rgb(44, 218, 157)";
export const filterButtonBorderColor = "rgb(44, 218, 157)";
export const filterButtonTextColor = "rgb(44, 218, 157)";

// EVENT LIST

export const eventListHeaderBgColor = "rgb(216, 216, 216)";
export const eventListHeaderColor = "rgb(51, 51, 51)";
export const eventCardTextColor = "rgb(45, 47, 127)";
export const eventPriceColor = "rgb(255, 255, 255)";
export const eventCardShadow = "rgba(0, 0, 0, 0.2)";
export const eventPriceBgColor = "rgb(45, 47, 127)";
export const sectionHeaderBgColor = "rgb(243, 243, 243)";
export const sectionHeaderShadow = "rgba(0, 0, 0, 0.14)";

// EVENT DETAILS
export const eventDetailsBgColor = "#FFF";
export const eventDetailsHeaderBgColor = "#F2F2F2";
export const eventCategoryLabelBgColor = "#D8D8D8";
export const eventDetailsBuyButtonColor = "#D8D8D8";
export const eventDetailsBuyButtonTextColor = "#FFFFFF";
