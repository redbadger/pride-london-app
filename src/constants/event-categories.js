// Ideally the keys here should be consts, but currently these are setup in
// contentful only as the label, hence we need to use this for the key too.
// If we move to realise categories as its own content type then this will
// go away and we will fetch this data.

const categories = {
  "Cabaret and Variety": {
    label: "Cabaret and Variety"
  },
  Community: {
    label: "Community"
  },
  "Talks and Debates": {
    label: "Talks and Debates"
  },
  "Film and Screenings": {
    label: "Film and Screenings"
  },
  "Plays and Theatre": {
    label: "Plays and Theatre"
  },
  "Social and Networking": {
    label: "Social and Networking"
  },
  Nightlife: {
    label: "Nightlife"
  },
  "Exhibition and Tours": {
    label: "Exhibition and Tours"
  },
  "Sports and Activities": {
    label: "Sports and Activities"
  },
  Health: {
    label: "Health"
  },
  Music: {
    label: "Music"
  }
};

export default categories;
