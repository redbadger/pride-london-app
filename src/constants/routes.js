export const EVENT_DETAILS = "EVENT_DETAILS";
export const EVENT_LIST = "EVENT_LIST";
export const FEATURED_EVENT_LIST = "FEATURED_EVENT_LIST";
export const HOME = "HOME";
export const EVENT_CATEGORIES_FILTER = "EVENT_CATEGORIES_FILTER";
export const PARADE = "PARADE";
export const SAVED = "SAVED";
export const SUPPORT_US = "SUPPORT_US";
export const FILTER_MODAL = "FILTER_MODAL";
export const DONATE = "DONATE";
export const SPONSOR = "SPONSOR";
