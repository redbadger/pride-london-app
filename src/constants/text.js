// @flow
const showEvents = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "No events";
  }

  if (numberOfEvents === 1) {
    return "Show 1 event";
  }

  return `Show ${numberOfEvents} events`;
};

const showEventsLabel = (numberOfEvents: number) => {
  if (numberOfEvents <= 0) {
    return "Show events (none selected)";
  }

  if (numberOfEvents === 1) {
    return "Show 1 selected event";
  }

  return `Show ${numberOfEvents} selected events`;
};

export const calendarTitleLabel = (dateRange: string, incomplete: boolean) =>
  `Selected: ${dateRange}${
    incomplete ? ", pick another day to select range" : ""
  }`;

export default {
  backButtonAccessibilityLabel: "Back",
  tabHome: "Home",
  tabEvents: "Events",
  tabParade: "Parade",
  tabSaved: "Saved",
  tabSupportUs: "Support us",
  tabParadeMap: "Map",
  tabParadeGroups: "Parade groups",
  featuredEventListTitle: "Featured events",
  filterByInterest: "All event types",
  categoryFilterButton: "Filter by event type",
  categoryFilterContents: "Filtering event type by",
  categoryFilterEmpty: "0 event types selected",
  addFilters: "Filter by area, price...",
  filters: "Filters",
  filterPickerApply: showEvents,
  filterPickerApplyLabel: showEventsLabel,
  filterDayPickerTitle: "Select dates",
  filterTimePickerTitle: "Select time",
  filterTitle: "Showing",
  eventDetailsPrice: "From: £",
  isFreePrice: "Free",
  eventFromPrice: "From",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Get tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets",
  selectDates: "Select dates",
  eventDetailsAbout: "About",
  eventDetailsSchedule: "Schedule",
  eventDetailsReadMore: "Read more",
  eventDetailsReadLess: "Read less",
  eventDetailsContactEmailSubject: "Pride In London event enquiry",
  eventDetailsContactEmailBody: "Dear event organiser,",
  collapsibleShowMore: "Show more",
  collapsibleShowLess: "Show less",
  savedEventsTitle: "Saved Events",
  supportUsTitle: "Support us",
  supportUsAsIndividual: "As an individual",
  supportUsAsBusiness: "As a business",
  supportUsDonate: "Donate",
  supportUsDonateDescription: "Every donation keeps us marching",
  supportUsShop: "Buy merchandise",
  supportUsShopDescription: "Visit our online shop on TheGayShop",
  supportUsSponsor: "Partner with us",
  supportUsSponsorDescription: "Support the community",
  supportUsVolunteer: "Volunteer",
  supportUsVolunteerDescription: "Passionate about equality? Join us.",
  sponsorTitle: "Partner with us",
  sponsorLevelHeading: "partners",
  sponsorHeadingOurPartners: "Our main 2018 partners",
  sponsorTextOurPartners:
    "A huge thank you to our main partners for their continued support.",
  sponsorHeadingPartnerWithUs: "Partner with us",
  sponsorTextPartnerWithUs:
    "Build your brand, engage your internal network and be part of a global movement to support diversity. To find out more, please contact sponsor@prideinlondon.org.",
  sponsorTextPartnerWithUsList:
    "* Pride in London is one of the top three biggest events in London, attracting an estimated one million visitors across the footprint. \n* Be part of our award winning marketing and advertising campaign. \n* 80% of those surveyed in our 2016 research agreed that they are more likely to think better of a brand if it is supportive of LGBT+ issues.",
  sponsorContactEmailSubject: "Sponsorship enquiry",
  sponsorContactButtonText: "Email to partner with us",
  donateTitle: "Donate",
  donateHeader: ["\"It's the first time I feel", 'accepted in a big group."'],
  donateIntroductionHeading: "Be part of the movement that drives diversity",
  donateIntroduction:
    "The need for Pride is as strong as ever. A huge amount of progress has been made for LGBT+ people but challenges and stigma are still rife. Whether you take part to celebrate, party or protest. Pride in London is your platform to make change. Your donations keep us marching.",
  donateAmountSelectionLabel: "Make a single donation",
  donateOtherAmountLabel: "Other amount (£)",
  donateMinimumAmount: "The minimum donation amount is £2.",
  donateButtonText: "Donate",
  anyDay: "Any day",
  anyTime: "Any time",
  time: {
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening"
  },
  tags: {
    audience: "Age group",
    price: "Price",
    free: "Show only free events",
    timeOfDay: "Time of day",
    morning: "Morning",
    afternoon: "Afternoon",
    evening: "Evening",
    venueDetails: "Venue options",
    accessibilityOptions: "Accessibility",
    "Gender neutral toilets": "Gender-neutral toilets",
    Indoors: "Indoor",
    Outdoors: "Outdoor",
    "Accessible Toilets": "Accessible toilets",
    "Wheelchair accessibile": "Wheelchair access",
    area: "Area of London"
  },
  cancel: "Cancel",
  allEvents: "All events",
  showEvents,
  showAllEvents: "Show all events",
  clearAll: "Clear all",
  reset: "Reset",
  filterEvents: "Filter Events",
  runsFrom: "Runs from:",
  noEventsTitle: "No events to show",
  noEventsDetails:
    "Try changing your filters to be less specific or check back later.",
  noSavedEventsTitle: "Where's the love?",
  noSavedEventsPart1: "No favourites yet? Tap the",
  noSavedEventsSaveButtonAlt: "'save this event' button",
  noSavedEventsPart2: "on any event you like to save them here.",
  noSavedEventsButton: "Find Events",
  saveEventButtonUnSaveEvent: "remove event from saved list",
  saveEventButtonSaveEvent: "save this event",
  homeViewAll: "View all",
  homeSupportUs: "Support us",
  homeSupportUsDescription: "Be part of the movement",
  parade: {
    headerTitle: "Parade - 7 July from 12pm"
  },
  paradeGroups: {
    title: "London Parade",
    subTitle: "Saturday 7 July from 12pm",
    paragraph1:
      "The Pride in London Parade is a platform for every part of London’s LGBT+ community. These groups, nearly 500 in total, represent our most diverse Pride in London yet. From LGBT+ parent groups and student services to sports clubs and national charities, we are proud to give a platform to our community representatives, supporters and allies.",
    paragraph2:
      "Find out who you can expect to see on the day from the list below.",
    linkLabelFacebook: "Facebook Page",
    linkLabelTwitter: "Twitter Profile",
    linkLabelWebsite: "Website",
    expandAccessibilityLabel: "Show more",
    collapseAccessibilityLabel: "Show less"
  },
  paradeInformationScreen: {
    pageHeading: "London Parade",
    pageSubheading: "Saturday 7 July",
    pageDescription:
      "The Pride in London Parade provides a platform for every part of London’s LGBT+ community to raise awareness of LGBT+ issues and campaign for the freedoms that will allow us to live our lives on a genuinely equal footing. It gives us a chance to be visible and speak loudly to the rest of the world about what we have achieved, how far we have come and what is still needed.",
    stages: [
      {
        stageHeading: "Trafalgar Square",
        stageSubheading: "",
        stageDescription:
          "One of the most popular and iconic areas on the day and close to where the Parade finishes. As usual there will be a mixture of famous faces and community acts from across the LGBT+ spectrum sprinkled with celebrities together with the Pride's Got Talent Finalists all under the watchful eye of Nelson's Column.",
        stageImage: "trafalgar"
      },
      {
        stageHeading: "Cabaret Stage",
        stageSubheading: "Dean Street",
        stageDescription:
          "Located in the Heart of Soho - Dean Street - no one should miss the Cabaret Stage, packed with diverse, colourful and entertaining acts from the LGBT+ Cabaret Community. This stage will also showcase the Pride's Got Talent Cabaret Finalists.",
        stageImage: "cabaret"
      },
      {
        stageHeading: "Women's / Diva Stage",
        stageSubheading: "Leicester Square",
        stageDescription:
          "Nestled in the iconic Leicester Square - this collaboration with Diva brings the best acts offered by and for the women's community. Expect high energy, fabulous talent and our female led Pride's Got Talent Acts.",
        stageImage: "diva"
      },
      {
        stageHeading: "Family Stage",
        stageSubheading: "Golden Square",
        stageDescription:
          "An intimate and fun stage, the best destination for fun and family entertainment.",
        stageImage: "family"
      },
      {
        stageHeading: "Community Village",
        stageSubheading: "Soho Square",
        stageDescription:
          "Pride in London is proud to provide a platform for charity and not for profit groups in Soho Square. Pop along to find out more about some amazing charities and not for profit groups.",
        stageImage: "community"
      }
    ]
  },
  storeVersionDialog: {
    updateAvailable: "New Update Available",
    newVersionDescription:
      "A new version of the app is available from the store. Do you want to update now?",
    askMeLater: "Ask me later",
    cancel: "Cancel",
    ok: "OK"
  },
  resetAllFilters: "Reset all filters"
};
