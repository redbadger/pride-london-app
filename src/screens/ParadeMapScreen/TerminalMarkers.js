// @flow
import React, { Fragment } from "react";
import { View, StyleSheet } from "react-native";
import { Marker } from "react-native-maps";
import Text, { scaleWithFont } from "../../components/Text";
import { lightNavyBlueColor, transparent } from "../../constants/colors";
import type {
  Terminals,
  Coordinates
} from "../../constants/parade-coordinates";

type Props = {
  terminals: Terminals[],
  onMarkerPress: () => void,
  onMarkerSelect: (event: { nativeEvent: { coordinate: Coordinates } }) => void
};

const TerminalMarkers = ({
  terminals,
  onMarkerPress,
  onMarkerSelect
}: Props) => {
  if (terminals.length === 0) {
    return null;
  }

  return (
    <Fragment>
      {terminals.map(terminal => (
        <Marker
          coordinate={terminal.coordinates}
          key={terminal.key}
          centerOffset={{ x: 0, y: -15 }}
          stopPropagation
          onPress={onMarkerPress}
          onSelect={onMarkerSelect}
        >
          <View style={styles.markerView}>
            <View style={styles.markerTextWrapper}>
              <Text type="h4" color="whiteColor">
                {terminal.text}
              </Text>
            </View>
            <View style={styles.triangle} />
          </View>
        </Marker>
      ))}
    </Fragment>
  );
};

const styles = StyleSheet.create({
  markerView: {
    alignItems: "center",
    justifyContent: "center"
  },
  markerTextWrapper: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: lightNavyBlueColor,
    height: Math.max(30, scaleWithFont("h4", 30)),
    width: Math.max(60, scaleWithFont("h4", 60)),
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 2
  },
  triangle: {
    width: 0,
    height: 0,
    backgroundColor: transparent,
    borderStyle: "solid",
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderBottomWidth: 10,
    borderLeftColor: transparent,
    borderRightColor: transparent,
    borderBottomColor: lightNavyBlueColor,
    transform: [{ rotate: "180deg" }]
  }
});

export default TerminalMarkers;
