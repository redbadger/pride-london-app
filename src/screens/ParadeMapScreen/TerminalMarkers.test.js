// @flow
import React from "react";
import { shallow } from "enzyme";
import { Marker } from "react-native-maps";
import TerminalMarkers from "./TerminalMarkers";
import { terminals } from "../../constants/parade-coordinates";

describe("TerminalMarkers component", () => {
  it("renders correctly", () => {
    const output = shallow(
      <TerminalMarkers
        terminals={terminals}
        onMarkerPress={() => {}}
        onMarkerSelect={() => {}}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders nothing when no terminals provided", () => {
    const output = shallow(
      <TerminalMarkers
        terminals={[]}
        onMarkerPress={() => {}}
        onMarkerSelect={() => {}}
      />
    );
    expect(output.children().length).toBe(0);
  });

  it("calls onMarkerPress function when pressed", () => {
    const mockHandleMarkerPress = jest.fn();
    const output = shallow(
      <TerminalMarkers
        terminals={terminals}
        onMarkerPress={mockHandleMarkerPress}
        onMarkerSelect={() => {}}
      />
    );

    output
      .find(Marker)
      .first()
      .simulate("press");

    expect(mockHandleMarkerPress).toHaveBeenCalled();
  });
});
