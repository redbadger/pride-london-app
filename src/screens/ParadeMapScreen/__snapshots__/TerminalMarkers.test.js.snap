// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminalMarkers component renders correctly 1`] = `
<React.Fragment>
  <MapMarker
    centerOffset={
      Object {
        "x": 0,
        "y": -15,
      }
    }
    coordinate={
      Object {
        "latitude": 51.51616,
        "longitude": -0.14223,
      }
    }
    key="start"
    onPress={[Function]}
    onSelect={[Function]}
    stopPropagation={true}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2d2f7f",
            "borderRadius": 2,
            "height": 30,
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "paddingVertical": 2,
            "width": 60,
          }
        }
      >
        <Text
          color="whiteColor"
          markdown={false}
          markdownStyle={Object {}}
          type="h4"
        >
          Start
        </Text>
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#2d2f7f",
            "borderBottomWidth": 10,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 5,
            "borderRightColor": "transparent",
            "borderRightWidth": 5,
            "borderStyle": "solid",
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
            "width": 0,
          }
        }
      />
    </Component>
  </MapMarker>
  <MapMarker
    centerOffset={
      Object {
        "x": 0,
        "y": -15,
      }
    }
    coordinate={
      Object {
        "latitude": 51.50497,
        "longitude": -0.12664,
      }
    }
    key="finish"
    onPress={[Function]}
    onSelect={[Function]}
    stopPropagation={true}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2d2f7f",
            "borderRadius": 2,
            "height": 30,
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "paddingVertical": 2,
            "width": 60,
          }
        }
      >
        <Text
          color="whiteColor"
          markdown={false}
          markdownStyle={Object {}}
          type="h4"
        >
          Finish
        </Text>
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#2d2f7f",
            "borderBottomWidth": 10,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 5,
            "borderRightColor": "transparent",
            "borderRightWidth": 5,
            "borderStyle": "solid",
            "height": 0,
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
            "width": 0,
          }
        }
      />
    </Component>
  </MapMarker>
</React.Fragment>
`;
